### ISSUE TITLE
MONITORON not working after hours

### ISSUE THREAD

**Issue by nixmeer** [Jopyth/MMM-Remote-Control#111]
Hello,

I run MMM on a raspberry pi zero w including chromium. As mentioned [here](https://github.com/Jopyth/MMM-Remote-Control/issues/90#issuecomment-424090387), I had to replace the executed command to turn the monitor on and off. It works perfectly fine for about 6 hours, and it won't work later on. It seems as if no command is working, at least REFRESH and MONITOROFF. For the past months, I rebooted the Pi every 6 hours which is a nasty solution.

The routine still seems to be called since I receive an answer. When I connect to the pi using ssh and type in the command that is being used by the MONITORON command, it works.

Any idea?

Best

Jan

**Comment by nixmeer** [comment]
So I think I can clarify the problem a little: 

After a couple of hours, the MONITORON and MONITOROFF commands are still being called. I verified this adding a line `res.send("1")` or `res.send("0")` and this is still done. I think, the following line [598](https://github.com/Jopyth/MMM-Remote-Control/blob/3ad512777f5f9cec0861950ff26ac1ad56fa4465/node_helper.js#L598) of node_helper.js is not correctly called, since the bash command itself works via ssh. In my case, it is `vcgencmd display_power 1` and `vcgencmd display_power 0`. 

Still I don't know what's wrong.

**Comment by max-8** [comment]
nixmeer,
I was having the same/similar issue after the new Remote-Control code release, and I am using the vcgencmd also (on a Pi Zero W).  I found that the new new code always test for monitor status before attempting to turn ON the monitor.  By default the status test is using "tvservice" command.  But tvservice will not report correctly if the monitor was turned off with the "vcgencmd display_power 0" command.  To make it work correctly I had to set up all 3 monitor customCommand's.

I deleted the entire MMM-Remote-Control directory and loaded fresh with a git pull.  Then added the below into my config.sys:

 {
  module: "MMM-Remote-Control",
        config: {
                     customCommand: {
                           monitorOnCommand: 'vcgencmd display_power 1',
                           monitorOffCommand: 'vcgencmd display_power 0',
                           monitorStatusCommand: 'vcgencmd display_power | grep -q "1" && echo "true" || echo "false" '
}}},

Everything has been working good since then.  Maybe this will work for you also.

**Comment by Hobohome** [comment]
@max-8  - Thanks! That resolved my issue with the monitor turning off on command, but not turning back on again.
Cheers

**Comment by nixmeer** [comment]
Hello :) 

thank you, @max-8 !! I got it working!

There is one more thing: When I tried to find a workaround for this "bug" (which might be a bug only on a Pi Zero W), I dived into the code and added HOMEBRIDGE_MONITORON, HOMEBRIDGE_MONITOROFF and HOMEBRIDGE_MONITORSTATUS which I would let return either "0" or "1" in order to work with a Homebridge plugin (Homebridge builds the bridge to Apple's HomeKit and Siri). The requirements are listed [here](https://github.com/jnovack/homebridge-better-http-rgb#interfacing). Since there is only a status, the reply is only 0 and 1. 

Any idea how to add this? Or where I have to change it? I don't know, how the js code executes these custom commands.

Thanks in advance :)
